### PR TITLE
Update schnorrkel with versioning

### PR DESCRIPTION
**Pull request by NikVolf**
Goes after #5328 

This will allow verifying old schnorrkel signatures using old runtimes
But will disallow producing new blocks and validating transactions with old signatures

### CODE REVIEW

File: primitives/core/src/sr25519.rs [hunk 1]
```diff
@@ -530,8 +561,15 @@ impl TraitPair for Pair {
 	}
 
 	/// Verify a signature on a message. Returns true if the signature is good.
-	fn verify<M: AsRef<[u8]>>(sig: &Self::Signature, message: M, pubkey: &Self::Public) -> bool {
-		Self::verify_weak(&sig.0[..], message, pubkey)
+	fn verify_format<M: AsRef<[u8]>>(format: Self::Format, sig: &Self::Signature, message: M, pubkey: &Self::Public) -> bool {
```
Comment (bkchr) [inline primitives/core/src/sr25519.rs]: I think we should not do this. We should really prevent that this type by default can verify an old signature.

I think we should only support checking the old signature in the implementation of the old `sr25519_verify` function.

Comment (NikVolf) [reply primitives/core/src/sr25519.rs]: By default it will deny old signature, see `Format::default` returns `Format::V2`

**Comment by NikVolf** [comment]
This one finally working on Kusama!

File: primitives/core/src/sr25519.rs (hunk 1 above)
Comment (NikVolf) [reply primitives/core/src/sr25519.rs]: Yeah, you are right! Too much versioning 😅

File: primitives/core/src/sr25519.rs [hunk 2]
```diff
@@ -566,6 +579,28 @@ impl Pair {
 		let kp = mini_key.expand_to_keypair(ExpansionMode::Ed25519);
 		(Pair(kp), mini_key.to_bytes())
 	}
+
+	/// Verify a signature on a message. Returns true if the signature is good.
+	pub fn verify_format<M: AsRef<[u8]>>(format: Format, sig: &Signature, message: M, pubkey: &Public) -> bool {
```
Comment (bkchr) [inline primitives/core/src/sr25519.rs]: Why isn't that just called verify deprecated?

There will be no more "formats" and formats is also not the correct term. It was broken before and got fixed.

I would drop all this format stuff, that really isn't needed and don't brings any advantages.

File: primitives/core/src/sr25519.rs [hunk 3]
```diff
@@ -787,7 +801,7 @@ mod test {
 		let js_signature = Signature::from_raw(hex!(
 			"28a854d54903e056f89581c691c1f7d2ff39f8f896c9e9c22475e60902cc2b3547199e0e91fa32902028f2ca2355e8cdd16cfe19ba5e8b658c94aa80f3b81a00"
 		));
-		assert!(Pair::verify(&js_signature, b"SUBSTRATE", &public));
+		assert!(Pair::verify_deprecated(&js_signature, b"SUBSTRATE", &public));
```
Comment (bkchr) [inline primitives/core/src/sr25519.rs]: Could you also check that verify fails with the normal verify function?

File: primitives/core/src/sr25519.rs [hunk 4]
```diff
@@ -609,14 +623,14 @@ mod compatibility_test {
...
-	fn verify_known_message_should_work() {
+	fn verify_known_old_message_should_work() {
 		let public = Public::from_raw(hex!("b4bfa1f7a5166695eb75299fd1c4c03ea212871c342f2c5dfea0902b2c246918"));
 		// signature generated by the 1.1 version with the same ^^ public key.
 		let signature = Signature::from_raw(hex!(
 			"5a9755f069939f45d96aaf125cf5ce7ba1db998686f87f2fb3cbdea922078741a73891ba265f70c31436e18a9acd14d189d73c12317ab6c313285cd938453202"
 		));
 		let message = b"Verifying that I am the owner of 5G9hQLdsKQswNPgB499DeA5PkFBbgkLPJWkkS6FAM6xGQ8xD. Hash: 221455a3\n";
-		assert!(Pair::verify(&signature, &message[..], &public));
+		assert!(Pair::verify_deprecated(&signature, &message[..], &public));
```
Comment (bkchr) [inline primitives/core/src/sr25519.rs]: Same.

**Comment by gavofyork** [comment]
oops... looks like i merged this too soon. please revert if needed.

**Pull request merged**